### DISCUSSION TITLE
DBeaver 7.3.0 - MySQL 5.7 ROW_COUNT() returns -1

### DISCUSSION THREAD

**Discussion by bsherwoodofdaptiv** [dbeaver/dbeaver#11138] [Q&A] [answered]
#### System information: 
- Operating system (distribution) and version
 - Ubuntu Linux - 20.04
- DBeaver version
 - 7.3.0
- Additional extensions
 - None

#### Connection specification:
- Database name and version
 -  MySQL 5.7.23
- Driver name
 - MySQL 5
- Do you use tunnels or proxies (SSH, SOCKS, etc)? 
 - No. Issues replicates on localhost

#### Describe the problem you're observing:
When using the ROW_COUNT() function after INSERT/UPDATE or SELECT, it always returns -1

This behavior is not seen with MySQL workbench to the same server and database

Connection configuration is as generic as possible. Out-Of-Box experience.

As behaving, DBeaver can not be used for any scripts that require using those values in logic.

#### Steps to reproduce, if exist:

Use any database table. Script it out to recreate it and edit script to give new table name. Run script to create empty destination table for testing.  Copy data from one table to the other, using the ROW_COUNT() function immediately afterward.

Example script that has the issue on my server:

insert into OLDER_EVENT 
select * from OLD_EVENT where EVENT_ID is not NULL ;
select ROW_COUNT();

Results (for my table)
- DBeaver: -1
- MySQL Workbench: 30   (which is the actual number of rows copied)

Second example:

delete from OLDER_EVENT;
select ROW_COUNT();

Results (for my table)
- DBeaver: -1
- MySQL Workbench: 30   (which is the actual number of rows deleted)

#### Include any warning/errors/backtraces from the logs

No errors generated. This is a behavior issue.

**Comment by bsherwoodofdaptiv** [comment]
Alternate but still undesired behavior: By using variables, the variable is set to "0". Example below:
`select @mycount := 1;

delete from OLDER_EVENT;
select @mycount := ROW_COUNT();
select @mycount;

insert into OLDER_EVENT 
select * from OLD_EVENT where EVENT_ID is not NULL ;
select @mycount := ROW_COUNT();
select @mycount;`

result for every select after the first variable assignment is "0"
In MySQL Workbench, the result is correct (as in first example in case, the count is 30 and shows as such)

**Comment by ShadelessFox** [comment, marked as answer]
Hello @bsherwoodofdaptiv,

I don't think `ROW_COUNT` function should be used in this case, since DBeaver can display statistics of your executed query, including the total number of updated rows. If you really want to use this function, you can execute your queries as a _script_, by using `Ctrl+E` shortcut.

Please see our wiki: https://github.com/dbeaver/dbeaver/wiki/SQL-Execution
